validator/accounts/account.go

```diff
@@ -254,6 +254,32 @@ func Merge(ctx context.Context, sourceDirectories []string, targetDirectory stri
 	return nil
 }
 
+// Split splits data from one validator database in sourceDirectory into several validator databases.
+// Each validator database is created in its own subdirectory inside targetDirectory.
+func Split(ctx context.Context, sourceDirectory string, targetDirectory string) (err error) {
+	var sourceStore *db.Store
+	sourceStore, err = db.GetKVStore(sourceDirectory)
+	if err != nil {
+		return errors.Wrap(err, "Failed to prepare the source database for splitting")
+	}
+	if sourceStore == nil {
+		return errors.New("no database found in source directory")
+	}
+	defer func() {
+		if sourceStore != nil {
+			if deferErr := sourceStore.Close(); deferErr != nil {
+				err = errors.Wrap(deferErr, "Failed to close the source database")
+			}
+		}
+	}()
+
+	if err := db.Split(ctx, sourceStore, targetDirectory); err != nil {
```

Comment:
I would just return db.Split here

Reply:
Doing this would lose the error message. Is that OK? I can't move the message up the stack because the caller of `Split` can't know whether it is the actual split that failed (there are other errors returned from the function).

Reply:
in the defer you can add a check before wrapping the error. if its non empty, you can wrap the error instead of overwriting it with the source close error.